runtime/src/accounts_hash.rs

```diff
@@ -554,188 +501,190 @@ impl AccountsHash {
         a.pubkey.partial_cmp(&b.pubkey).unwrap()
     }
 
-    fn sort_hash_intermediate(
-        data_by_pubkey: Vec<Vec<CalculateHashIntermediate>>,
-        stats: &mut HashStats,
-    ) -> Vec<Vec<CalculateHashIntermediate>> {
-        // sort each PUBKEY_DIVISION vec
-        let mut sort_time = Measure::start("sort");
-        let sorted_data_by_pubkey: Vec<Vec<_>> = data_by_pubkey
-            .into_par_iter()
-            .map(|mut pk_range| {
-                // has to be a stable sort because items are in slot order already
-                pk_range.sort_by(Self::compare_two_hash_entries);
-                pk_range
-            })
-            .collect();
-        sort_time.stop();
-        stats.sort_time_total_us += sort_time.as_us();
-        sorted_data_by_pubkey
-    }
-
     pub fn checked_cast_for_capitalization(balance: u128) -> u64 {
         balance
             .try_into()
             .expect("overflow is detected while summing capitalization")
     }
 
     fn de_dup_and_eliminate_zeros(
-        sorted_data_by_pubkey: Vec<Vec<CalculateHashIntermediate>>,
+        sorted_data_by_pubkey: Vec<Vec<Vec<CalculateHashIntermediate>>>,
         stats: &mut HashStats,
-    ) -> (Vec<Vec<Vec<Hash>>>, u64) {
+        range: &Range<usize>,
+    ) -> (Vec<Vec<Hash>>, u64) {
         // 1. eliminate zero lamport accounts
         // 2. pick the highest slot or (slot = and highest version) of each pubkey
         // 3. produce this output:
-        // vec: PUBKEY_BINS_FOR_CALCULATING_HASHES in pubkey order
-        //   vec: sorted sections from parallelism, in pubkey order
-        //     vec: individual hashes in pubkey order
+        // a. vec: PUBKEY_BINS_FOR_CALCULATING_HASHES in pubkey order
+        //      vec: individual hashes in pubkey order, 1 hash per
+        // b. lamports
         let mut zeros = Measure::start("eliminate zeros");
         let overall_sum = Mutex::new(0u64);
-        const CHUNKS: usize = 10;
-        let hashes: Vec<Vec<Vec<Hash>>> = sorted_data_by_pubkey
+        let unreduced_entries = AtomicUsize::new(0);
+        let hashes: Vec<Vec<Hash>> = (range.start..range.end)
             .into_par_iter()
-            .map(|pubkey_division| {
-                let (hashes, sum) = Self::de_dup_accounts_in_parallel(&pubkey_division, CHUNKS);
+            .map(|bin| {
+                let (hashes, sum, unreduced_entries_count) =
+                    Self::de_dup_accounts_in_parallel(&sorted_data_by_pubkey, bin);
                 let mut overall = overall_sum.lock().unwrap();
                 *overall = Self::checked_cast_for_capitalization(sum as u128 + *overall as u128);
+                unreduced_entries.fetch_add(unreduced_entries_count, Ordering::Relaxed);
```

Comment:
Probably small, but does it need to be in the `overall_sum` critical section?

Reply:
Do you mean the unreduced_entries.fetch_add occurring while overall_sum is still locked?
No, it doesn't have to be like that. I can reverse them to avoid that.